integration-libs/cdc/schematics/add-cdc/index.ts

```diff
@@ -33,37 +30,38 @@ import {
 } from '../constants';
 
 export function addCdcFeature(options: SpartacusCdcOptions): Rule {
-  return (tree: Tree, _context: SchematicContext) => {
+  return (tree: Tree, context: SchematicContext) => {
     const packageJson = readPackageJson(tree);
     validateSpartacusInstallation(packageJson);
 
     return chain([
-      addCdc(options),
+      shouldAddFeature(CLI_CDC_FEATURE, options.features)
+        ? addCdc(options)
+        : noop(),
 
-      addCdcPackageJsonDependencies(packageJson),
-      installPackageJsonDependencies(),
+      addCdcPackageJsonDependencies(packageJson, context, options),
     ]);
   };
 }
 
-function addCdcPackageJsonDependencies(packageJson: any): Rule {
-  const spartacusVersion = `^${getSpartacusSchematicsVersion()}`;
-  const spartacusDependencies: NodeDependency[] = [
-    {
-      type: NodeDependencyType.Default,
-      version: spartacusVersion,
-      name: SPARTACUS_ASM,
-    },
-    {
-      type: NodeDependencyType.Default,
-      version: spartacusVersion,
-      name: SPARTACUS_USER,
-    },
-  ];
+function addCdcPackageJsonDependencies(
+  packageJson: any,
+  context: SchematicContext,
+  options: SpartacusCdcOptions
+): Rule {
+  const spartacusLibraries = createSpartacusDependencies(peerDependencies);
   const thirdPartyDependencies = createDependencies(peerDependencies);
+  const dependencies = spartacusLibraries.concat(thirdPartyDependencies);
+
+  const rule = addPackageJsonDependencies(dependencies, packageJson);
+
+  const featureOptions = createSpartacusFeatureOptionsForLibrary(
+    spartacusLibraries.map((dependency) => dependency.name),
+    options
+  );
+  addSchematicsTasks(featureOptions, context);
 
-  const dependencies = spartacusDependencies.concat(thirdPartyDependencies);
-  return addPackageJsonDependencies(dependencies, packageJson);
+  return rule;
```

Comment:
Can we use this code in all feature/integration libs? And wrap that in some handy util?
If I would add dependency in any feature lib to cross package I would like to not touch this code ever and have it normally installing cross package with external schematics